rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/action/TbAbstractAlarmNode.java

```diff
@@ -73,32 +73,33 @@ public void onMsg(TbContext ctx, TbMsg msg) {
 
     protected abstract ListenableFuture<TbAlarmResult> processAlarm(TbContext ctx, TbMsg msg);
 
-    protected ListenableFuture<JsonNode> buildAlarmDetails(TbMsg msg, JsonNode previousDetails) {
+    protected ListenableFuture<JsonNode> buildAlarmDetails(TbContext ctx, TbMsg msg, JsonNode previousDetails) {
         try {
             TbMsg dummyMsg = msg;
             if (previousDetails != null) {
                 TbMsgMetaData metaData = msg.getMetaData().copy();
                 metaData.putValue(PREV_ALARM_DETAILS, JacksonUtil.toString(previousDetails));
                 dummyMsg = TbMsg.transformMsgMetadata(msg, metaData);
             }
-            return scriptEngine.executeJsonAsync(dummyMsg);
+            ctx.logJsEvalRequest();
+            ListenableFuture<JsonNode> alarmDetailsFuture = scriptEngine.executeJsonAsync(dummyMsg);
```

Comment:
Consider improvement similar to the LogNode:
```
        if (standard) {
            logStandard(ctx, msg);
            return;
        }
```

Reply:
[c5938b1](https://github.com/thingsboard/thingsboard/pull/10547/commits/c5938b13393798c7c17c33ee50b7c5b21da71fea)